examples/misc/lib/language_tour/classes/point_this.dart

```diff
@@ -0,0 +1,16 @@
+// ignore_for_file: invalid_reference_to_this, unnecessary_this
+// #docregion this-late
+double initialX = 1.5;
+
+class Point {
+  double? x = initialX;
+  // OK, can access declarations that do not depend on `this`.
+  double? y = this.x;
+  // ERROR, can't access `this` in non-`late` initializer.
+  late double? z = this.x;
+  // OK, can access `this` in `late` initializer.
+
+  Point(this.x, this.y);
+  // OK, not an initializing expression, can access `this`.
```

Comment:
This one is difficult. ;-)

I think it's very useful to state explicitly that `this.x` and `this.y` are not expressions at all, not even the otherwise plausible construct that we call `<assignableExpression>` which is a specification of the target of an assignment (occurring on the left hand side of `=`).

If we had considered each of them as the left hand side of an assignment then we could be initializing an instance of a subclass of `Point` that declares `set x(int value) {...}`, and in that case we'd have to call that setter. That's not the actual semantics: `this.x` specifies that the actual argument is stored in the instance variable `x` in the current class body, it will never call an overriding setter. So `this.x` is not an expression and it's not an assignable expression.

In order to make that point even more firmly I think it's safer to avoid talking about "can access `this`" at all: that sounds a bit like "the semantics of `this.x` includes expression evaluation, and during that expression evaluation we can evaluate `this`, just like any other expression". And we don't want the reader to think like that.

Perhaps we could say something like this:

```suggestion
  // OK, `this.someName` is a parameter declaration, not an expression.
```